packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,154 @@
+    // is going somewhere else, it means we will lose whatever is on those outputs given a failure.
+    if (!tx.body.collateralReturn || !addresses.includes(tx.body.collateralReturn.address))
+      return totalOwnedValueAtRisk;
+
+    // Should never be negative in a correctly balanced transaction, but let's do a sanity check anyway.
+    return BigIntMath.max([totalOwnedValueAtRisk - tx.body.collateralReturn.value.coins, 0n]) ?? 0n;
+  }
+
+  // Legacy collaterals, we simply return the addition of the collateral inputs we own.
+  return totalOwnedValueAtRisk;
+};
+
+const totalInputsValue = (resolvedInputs: ResolutionResult) => {
+  const receivedInputsValues = resolvedInputs.resolvedInputs.map((input) => input.value);
+  return coalesceValueQuantities(receivedInputsValues);
+};
+
+const totalOutputsValue = (outputs: Cardano.TxOut[]) => coalesceValueQuantities(outputs.map((output) => output.value));
+
+const mintInspectionToTokenMap = (mintedAssets: AssetsMintedInspection) =>
+  new Map<Cardano.AssetId, bigint>(
+    mintedAssets.map((asset) => [AssetId.fromParts(asset.policyId, asset.assetName), asset.quantity])
+  );
+
+const getImplicitAssets = async (tx: Cardano.Tx) => {
+  const mintedAssets = mintInspectionToTokenMap(await assetsMintedInspector(tx));
+  const burnedAssets = mintInspectionToTokenMap(await assetsBurnedInspector(tx));
+
+  return coalesceTokenMaps([mintedAssets, burnedAssets]);
+};
+
+const getUnaccountedFunds = async (
+  tx: Cardano.Tx,
+  resolvedInputs: ResolutionResult,
+  implicitCoin: Cardano.Lovelace,
+  implicitAssets: Cardano.TokenMap = new Map()
+): Promise<Cardano.Value> => {
+  const totalInputs = totalInputsValue(resolvedInputs);
+  const totalOutputs = totalOutputsValue(tx.body.outputs);
+
+  totalInputs.assets = coalesceTokenMaps([totalInputs.assets, implicitAssets]);
+  totalInputs.coins += implicitCoin;
+
+  return subtractValueQuantities([totalOutputs, totalInputs]);
+};
+
+/**
+ * Inspects a transaction and produces a summary.
+ *
+ * @param {TransactionSummaryInspectorArgs} args The arguments for the inspector.
+ */
+export const transactionSummaryInspector: TransactionSummaryInspector =
+  (args: TransactionSummaryInspectorArgs) => async (tx) => {
+    const { inputResolver, addresses, rewardAccounts, protocolParameters } = args;
+    const resolvedInputs = await resolveInputs(tx.body.inputs, inputResolver);
+    const fee = tx.body.fee;
+
+    const implicit = computeImplicitCoin(
+      protocolParameters,
+      { certificates: tx.body.certificates, withdrawals: tx.body.withdrawals },
+      rewardAccounts || []
+    );
+
+    const collateral = await getCollateral(tx, inputResolver, addresses);
+
+    const totalOutputValue = await totalAddressOutputsValueInspector(addresses)(tx);
+    const totalInputValue = await totalAddressInputsValueInspector(addresses, inputResolver)(tx);
+    const implicitCoin = implicit.withdrawals! + implicit.reclaimDeposit! - implicit.deposit!;
```

Comment:
When would those be null/undefined? Shouldn't we use 0 as default then?
```suggestion
    const implicitCoin = implicit.withdrawals + implicit.reclaimDeposit - implicit.deposit;
```

Reply:
I didn't understand why they are optional either, they should be 0, this is the code where it is defined:

```
/** Implicit coin quantities used in the transaction */
export interface ImplicitCoin {
  /** Reward withdrawals */
  withdrawals?: Lovelace;
  /** Reward withdrawals + deposit reclaims (total return) */
  input?: Lovelace;
  /** Delegation registration deposit */
  deposit?: Lovelace;
  /** Deposits returned */
  reclaimDeposit?: Lovelace;
}

```
I removed the ? from the fields and broke the property testing and it was not obvious how to fix them for me 😅 , so I left them optional

Reply:
Got it, so I think the code in suggested change should do `|| 0n` for each of those

Reply:
done here: https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/76d140d23288335920f0074dd659f7bd5f148109